snakemake/__init__.py

```diff
@@ -2751,6 +2785,33 @@ def parse_cores(cores):
                 )
                 sys.exit(1)
 
+    if args.az_batch:
+        if not args.default_remote_provider or not args.default_remote_prefix:
+            print(
+                "Error: --az-batch must be combined with "
+                "--default-remote-provider AzBlob and --default-remote-prefix to "
+                "provide a blob container name\n",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+        elif args.az_batch_account_url is None:
+            print(
+                "Error: --az-batch-account-url must be set when --az-batch is used\n",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+        elif not url_can_parse(args.az_batch_account_url):
+            print(
+                "Error: invalide azure batch account url, please use format: https://{account_name}.{location}.batch.azure.com."
```

Comment:
What is the rationale for representing this as url instead of expecting a location and an account flag?

Reply:
Since the full batch_account_url is used as an argument to BatchServiceClient in line 165, and the batch_account_name is used for the SharedKeyCredentials in line 162, but the location is never used. 

It's a tradeoff I guess. Having it the other way would mean increasing the probability of user error when constructing the batch_account_url, if either the location or the name is mis-specified. This way someone only has to enter the correct value once. Also since location is not used, that flag would be an additional 5ish lines of code throughout the argument parsing logic that really is only used to construct the full batch_account_url and nothing else..